llvm/lib/SYCLLowerIR/ESIMD/ESIMDOptimizeVecArgCallConv.cpp

```diff
@@ -510,7 +510,7 @@ ESIMDOptimizeVecArgCallConvPass::run(Module &M, ModuleAnalysisManager &MAM) {
 
   for (Function &F : M) {
     const bool FReplaced = processFunction(&F);
-    Modified &= FReplaced;
+    Modified |= FReplaced;
```

Comment:
as a result of this error, Modified on return from this function was always 'false'